### DISCUSSION TITLE
Empty userHandle when trying to login using Android phone

### DISCUSSION THREAD

**Discussion by rashmidusane** [webauthn4j/webauthn4j#529] [Q&A]
Hello,

While authenticating using FIDO2 with android11 (Google Pixel), the browser(latest Chrome) sends back EMPTY userhandle during assertion verification. Because of this, login fails. Did anybody observe this issue?  Using USB Yubikey for testing.

Thanks!

**Comment by ynojima** [comment]
Returned `userHandle` may be null.
https://www.w3.org/TR/2019/REC-webauthn-1-20190304/#ref-for-public-key-credential-source-userhandle%E2%91%A1

Could you share the tracetrace you faced?

**Comment by rashmidusane** [comment]
Thank you very much for your reply!

No, the userHandle is not null, it's empty.  The credential which was sent
to the server looks like the following. (sorry, I have replaced data with
XX). On the server side, it fails because userHandle is missing. The same
code works well on Mac/windows Chrome but in the case of Android, this is
the issue.


   -

   {
     "id": "XXXXXXXXXXXXXXXXXXX",
     "rawId": "XXXXXXXXXXXXXXXXXXX",
     "type": "public-key",
     "response": {
       "authenticatorData": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
       "clientDataJSON":
"eyJ0eXBlIjoid2ViYXV0aG4uZ2V0IiwiY2hhbGxlbmdlIjoidnd3MDhBODdxaExmT2Ezb0VkYUdGNFhqZHhxc1FwZTZDZklfcWVwRGk0NCIsIm9yaWdpbiI6IlhYWFgiLCJhbmRyb2lkUGFja2FnZU5hbWUiOiJjb20uYW5kcm9pZC5jaHJvbWUiLCJhbGciOiJIUzI1NiJ9.e30.o4rQjGKVyW-6B55_YDS8Ky3IvNetsgDypF5vd222xoA",
       "signature": "XXXXXXXXXXXXXXXXXXX",
       "userHandle": ""
     }
   }

**Comment by ynojima** [comment]
Ummm, I couldn't reproduce it. I tried with latest Android Chrome, Security Key by Yubico, and Auth0 WebAuthn debugger, but it doesn't return empty array. Could you share the details how to reproduce it?

![Screenshot_20210701-074042](https://user-images.githubusercontent.com/2139672/124040821-a0795380-da40-11eb-9c64-724a2a246c9f.png)
